src/Fields/field.jl

```diff
@@ -245,6 +245,10 @@ function offset_windowed_data(data, Loc, grid, indices)
         windowed_parent = parent(data)
     else
         parent_indices = map(parent_index_range, indices, loc, topo, halo)
+        if size(data)[3] == 1 && first.(axes(data))[3] == grid.Nz + 1 # take ssh into consideration
```

Comment:
What is special about `3`? This code is not supposed to know anything about physics or which way is up...

Reply:
What does

```julia
first.(axes(data))[3] == grid.Nz + 1
```

mean?\

Reply:
It's a very specific conditional statement stating that if the size of third dimension is 1, and its first (only) index is Nz+1, then perform the following operation. But thanks to my discussion with you, @glwagner, the issue has now been resolved in commit [bbfc656](https://github.com/CliMA/Oceananigans.jl/pull/3573/commits/bbfc656582689e35256f0f3f8f93e98c2479c810).

Reply:
ok great, note too that `size(data, 3)` is preferred to `size(data)[3]`